spec/components/tab_navigation_component_spec.rb

```diff
@@ -24,8 +24,19 @@
   end
 
   context 'with link for current request' do
-    before do
-      allow(vc_test_request).to receive(:path).and_return('/first')
+    let(:request_path) { '/first' }
+
+    class ExampleController < ApplicationController; end
```

Comment:
the way RSpec uses blocks, class definitions like this end up at the global scope, so they can collide across specs, especially with a generic name like `ExampleController`

A few options:
1. Much more specific controller name `TabNavExampleController`
2. anonymous controller (`let(:example_controller_class) { Class.new(ApplicationController)`
    - this probably wouldn't play well with the router :[
3. Delete/Undef the controller class in after the block cleanup? `Object.remove_const(:ExampleController)`

Reply:
Good call-out. I know RSpec creates its own namespace for each spec, so maybe we can tie into that? Dunno if there's a programmatic way, but maybe it's fine to do manually. Based on some quick local testing:

```suggestion
    class RSpec::ExampleGroups::TabNavigationComponent::ExampleController < ApplicationController; end
```

Having some trouble with actually getting the routing to respect this, though (capitalization, etc).

Reply:
Actually, since we don't care about running any code associated with the controller action, and since [ViewComponent already assumes the presence of an `ApplicationController`](https://github.com/ViewComponent/view_component/blob/main/docs/guide/testing.md#configuring-the-controller-used-in-tests), maybe it's fine to not define a new controller at all, and use that instead.

See 219bd88b9

Reply:
even better option!